### ISSUE TITLE
Misleading warning when email doesn’t match account email

### ISSUE THREAD

**Issue by j-f1** [desktop/desktop#15795]
### The problem

When I change the local Git config for a repo to use an email other than my own, I get this warning:

> **This commit will be misattributed**
> The email in your global Git config ([_redacted_]) doesn't match your GitHub account.
> You can also choose an email local to this repository from the repository settings.

This is misleading:

- The email in my global Git config does match my GitHub account.
- The local email is the one at issue, so it should probably say that?

### Release version

3.1.3-beta3 (arm64)

### Operating system

macOS 12.6.1 (21G217)

### Steps to reproduce the behavior

1. Make sure your global Git email is set to an email connected to your GitHub account
2. In a Git repository, set your email to an email associated with a different account
3. Open GitHub desktop and click on the ⚠️ badge of the avatar

### Log files

[2022-12-16.desktop.beta.log](https://github.com/desktop/desktop/files/10247462/2022-12-16.desktop.beta.log)

**Comment by davidaucoin7377** [comment]
https://github.com/desktop/desktop/files/10247462/2022-12-16.desktop.beta.log

**Comment by steveward** [comment]
Thanks for reporting this @j-f1. I reproduced this behavior as well, and have marked it as a bug to get fixed.

**Comment by davidaucoin7377** [comment]
I'm trying to figure out where to start with claiming my ens on etherscan

**Comment by skyvondave** [comment]
I'm getting this same message, and the Learn More link is just confounding me more.

![image](https://user-images.githubusercontent.com/40671781/222943481-6667eda0-90d5-498a-859e-ca35e364f9e6.png)

The Learn More link: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-in-git

I realize this may be a bug, but... what intended situation would the error apply to? It seems like it has no reason to exist at all... the only options on that tab are "Use my global Git config" or "Use a local Git config". If I select the latter, why would I add the same email address as the global one? It seems the warning is about something that _legimately will happen_ that is undesired. The Learn More link is a huge blurb explaining the warning... all of that was accidental?

Now I'm concerned that my commits will actually be wrongly attributed... which is the case? How do we know which logic will happen? I suppose I can only test on my own or clients' repos...

I'd think multiple GitHub accounts would not be an issue, especially considering a lot of people's work is not just on one GitHub account on one computer (and it sounds silly to expect otherwise). I have many clients now, and there's no reason to have different computers with their own GitHub Desktop installs for each client that only allows their own GitHub users (which they create) to work on their projects. 

Sure that segregation may be some sort of ideal, but it's not realistic to have some sort of restriction to it. At least it appears GitHub is trying to do that, or just has some bad logic here, or an improper warning.

The app should be able to let me log into multiple GitHub accounts and let me assign repos to those. It can still set a global config user as a default, but also let me change the default, while also having a repo setting to smartly use a local user. I'm honestly surprised none of this appears to have even been thought about much when GitHub Desktop was being built. What blocks are there to a sensible multi-user configuration?

**Comment by ColinRoxs** [comment]
Yup still happening and to make things worse Github desktop is telling me to update my email to """Match""" my account's email. But the email it wants to update to is a default no reply email.

**Comment by christophfriedrich** [comment]
I had this today as well, that suddenly this triangle appeared and suggested to "match" my email address to a generic privacy-preserving `@noreply.github.com` address (sadly didn't get a screenshot of the exact dialogue). In the dropdown menu there was only this 1 noreply address to be selected, the 4 email addresses I have _actually_ connected to my account were not shown.

Then an hour or so later, I got this popup:

> Your account token has been invalidated and you have been signed out from your GitHub.com account. Do you want to sign in again?

<img width="624" height="211" alt="Image" src="https://github.com/user-attachments/assets/1ae6b60e-010b-48be-be90-ea8c2955f11f" />

Apparently the message was only displayed because the app was no longer able to read my actual email?
The claim that the configured email "doesn't match your GitHub account" was in fact wrong.
